cpmpy/solvers/exact.py

```diff
@@ -0,0 +1,492 @@
+                assert isinstance(sub_expr, Comparison), "Implication must have linear constraints on right hand side"
+
+                lhs, rhs = sub_expr.args
+                assert isinstance(lhs, _NumVarImpl) or lhs.name == "sum" or lhs.name == "wsum",f"Unknown linear expression {lhs} on right side of indicator constraint: {cpm_expr}"
+
+                xct_coefs, xct_vars, xct_rhs = self._make_numexpr(lhs,rhs)
+
+                if isinstance(cond, NegBoolView):
+                    cond, bool_val = self.solver_var_no_num(cond._bv), False
+                else:
+                    cond, bool_val = self.solver_var_no_num(cond), True
+
+                if sub_expr.name == "==":
+                    if bool_val:
+                        # a -> b==c
+                        # a -> b>=c and a -> -b>=-c
+                        self._add_xct_reif_right(cond, xct_coefs,xct_vars,xct_rhs)
+                        self._add_xct_reif_right(cond, [-x for x in xct_coefs],xct_vars,-xct_rhs)
+                    else:
+                        # !a -> b==c
+                        # !a -> b>=c and !a -> -b>=-c
+                        #  a <- b<c  and  a <- -b<-c
+                        #  a <- -b>=1-c  and  a <- b>=1+c
+                        self._add_xct_reif_left(cond, xct_coefs,xct_vars,1+xct_rhs)
+                        self._add_xct_reif_left(cond, [-x for x in xct_coefs],xct_vars,1-xct_rhs)
+                elif sub_expr.name == ">=":
+                    if bool_val:
+                        # a -> b >= c
+                        self._add_xct_reif_right(cond, xct_coefs,xct_vars,xct_rhs)
+                    else:
+                        # !a -> b>=c
+                        #  a <- b<c
+                        #  a <- -b>=1-c
+                        self._add_xct_reif_left(cond, [-x for x in xct_coefs],xct_vars,1-xct_rhs)
+                elif sub_expr.name == "<=":
+                    if bool_val:
+                        # a -> b=<c
+                        # a -> -b>=-c
+                        self._add_xct_reif_right(cond, [-x for x in xct_coefs],xct_vars,-xct_rhs)
+                    else:
+                        # !a -> b=<c
+                        #  a <- b>c
+                        #  a <- b>=1+c
+                        self._add_xct_reif_left(cond, xct_coefs,xct_vars,1+xct_rhs)
+                else:
+                    raise NotImplementedError(
+                    "Unexpected condition constraint for Exact '{}' {}".format(lhs.name,cpm_expr))
+
+            # True or False
+            elif isinstance(cpm_expr, BoolVal):
+                self._add_xct_constr([], [], True, 0 if cpm_expr.args[0] else 1, False, 0)
+
+            # a direct constraint, pass to solver
+            elif isinstance(cpm_expr, DirectConstraint):
+                assert False, "Exact does not support direct constraints"
+
+            else:
+                raise NotImplementedError(cpm_expr)  # if you reach this... please report on github
+
```

Comment:
I think you need to return self, I think one test is failing for that (e.g. s += con; s becoming None)

Reply:
Done